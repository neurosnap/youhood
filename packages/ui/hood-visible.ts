import * as h from 'react-hyperscript';
import { connect } from 'react-redux';
import styled from 'styled-components';

import { actions } from '@youhood/hood';
const { showAllHoods, hideAllHoods } = actions;

import { HoodBarButton } from './ui';

const Visible = styled.div`
  display: flex;
  margin-left: 0;
  position: relative;
  height: 100%;
`;

const noop = () => {};

interface VisibleProp {
  (): void;
}

interface Props {
  showAll: VisibleProp;
  hideAll: VisibleProp;
}

export const HoodVisible: React.SFC<Props> = ({ showAll, hideAll }) =>
  h(Visible, [
    h(HoodBarButton, { onClick: showAll }, 'Show Hoods'),
    h(HoodBarButton, { onClick: hideAll }, 'Hide Hoods'),
  ]);

HoodVisible.defaultProps = {
  showAll: noop,
  hideAll: noop,
};

export default connect(
  null,
  (dispatch) => ({
    showAll: () => dispatch(showAllHoods()),
    hideAll: () => dispatch(hideAllHoods()),
  }),
)(HoodVisible as any);
