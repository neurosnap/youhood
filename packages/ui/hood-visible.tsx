import * as React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';

import { actions } from '@youhood/hood';
const { showAllHoods, hideAllHoods, showOnlyWinnerHoods } = actions;

import { HoodBarButton } from './ui';

const Visible = styled.div`
  display: flex;
  margin-left: 0;
  position: relative;
  height: 100%;
`;

const noop = () => {};

type VisibleProp = () => void;
interface Props {
  showAll: VisibleProp;
  hideAll: VisibleProp;
  showWinners: VisibleProp;
}

export const HoodVisible: React.SFC<Props> = ({
  showAll,
  hideAll,
  showWinners,
}) => (
  <Visible>
    <HoodBarButton onClick={showWinners}>Show Current Hoods</HoodBarButton>
    <HoodBarButton onClick={showAll}>Show All Hoods</HoodBarButton>
    <HoodBarButton onClick={hideAll}>Hide All Hoods</HoodBarButton>
  </Visible>
);

HoodVisible.defaultProps = {
  showAll: noop,
  hideAll: noop,
};

export default connect(
  null,
  (dispatch) => ({
    showAll: () => dispatch(showAllHoods()),
    hideAll: () => dispatch(hideAllHoods()),
    showWinners: () => dispatch(showOnlyWinnerHoods()),
  }),
)(HoodVisible);
