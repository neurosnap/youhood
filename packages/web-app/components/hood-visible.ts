import * as h from 'react-hyperscript';
import { connect } from 'react-redux';
import styled from 'styled-components';

import { actionCreators } from '@youhood/hood';
const { showAllHoods, hideAllHoods } = actionCreators;

import { HoodBarButton } from './ui';

const Visible = styled.div`
  display: flex;
  margin-left: 0;
  position: relative;
  height: 100%;
`;

interface VisibleProp {
  (): void;
}

interface Props {
  showAll: VisibleProp;
  hideAll: VisibleProp;
}

export const HoodVisible = ({ showAll, hideAll }: Props) => h(Visible, [
  h(HoodBarButton, { onClick: showAll }, 'Show Hoods'),
  h(HoodBarButton, { onClick: hideAll }, 'Hide Hoods'),
]);

export default connect(null, (dispatch) => ({
  showAll: () => dispatch(showAllHoods()),
  hideAll: () => dispatch(hideAllHoods()),
}))(HoodVisible as any);

