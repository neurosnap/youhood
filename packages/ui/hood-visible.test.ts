import { mount } from 'enzyme';
import * as h from 'react-hyperscript';

import { HoodVisible } from './hood-visible';

describe('HoodVisible', () => {
  it('should render the buttons', () => {
    const tree = mount(h(HoodVisible));
    expect(tree).toMatchSnapshot();
  });
});
