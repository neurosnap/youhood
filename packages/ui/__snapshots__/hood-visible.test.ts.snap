// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HoodVisible should render the buttons 1`] = `
<Component
  hideAll={[Function]}
  showAll={[Function]}
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-csuQGl",
            "isStatic": true,
            "lastClassName": "gejFaP",
            "rules": Array [
              "
  display: flex;
  margin-left: 0;
  position: relative;
  height: 100%;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-csuQGl",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="gejFaP"
      >
        <Styled(styled.button)>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-eNQAEJ",
                  "isStatic": true,
                  "lastClassName": "bmXkwR",
                  "rules": Array [
                    "
  cursor: pointer;
  height: 100%;
  display: flex;
  align-items: center;
  padding: 0 0.4rem;
  margin: 0 0.4rem;
  background: inherit;
  color: inherit;
  border: none;
  font: inherit;
  cursor: pointer;
  outline: inherit;

  :hover:enabled {
    background-color: ",
                    "rgba(255, 255, 255, 0.25)",
                    ";
  }

  :disabled {
    color: ",
                    "#ccc",
                    ";
  }
",
                    "
  font-size: ",
                    "0.8rem",
                    ";
",
                  ],
                },
                "displayName": "Styled(styled.button)",
                "foldedComponentIds": Array [
                  "sc-jKJlTe",
                ],
                "render": [Function],
                "styledComponentId": "sc-eNQAEJ",
                "target": "button",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <button
              className="bmXkwR"
            >
              Show Current Hoods
            </button>
          </StyledComponent>
        </Styled(styled.button)>
        <Styled(styled.button)
          onClick={[Function]}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-eNQAEJ",
                  "isStatic": true,
                  "lastClassName": "bmXkwR",
                  "rules": Array [
                    "
  cursor: pointer;
  height: 100%;
  display: flex;
  align-items: center;
  padding: 0 0.4rem;
  margin: 0 0.4rem;
  background: inherit;
  color: inherit;
  border: none;
  font: inherit;
  cursor: pointer;
  outline: inherit;

  :hover:enabled {
    background-color: ",
                    "rgba(255, 255, 255, 0.25)",
                    ";
  }

  :disabled {
    color: ",
                    "#ccc",
                    ";
  }
",
                    "
  font-size: ",
                    "0.8rem",
                    ";
",
                  ],
                },
                "displayName": "Styled(styled.button)",
                "foldedComponentIds": Array [
                  "sc-jKJlTe",
                ],
                "render": [Function],
                "styledComponentId": "sc-eNQAEJ",
                "target": "button",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            onClick={[Function]}
          >
            <button
              className="bmXkwR"
              onClick={[Function]}
            >
              Show All Hoods
            </button>
          </StyledComponent>
        </Styled(styled.button)>
        <Styled(styled.button)
          onClick={[Function]}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-eNQAEJ",
                  "isStatic": true,
                  "lastClassName": "bmXkwR",
                  "rules": Array [
                    "
  cursor: pointer;
  height: 100%;
  display: flex;
  align-items: center;
  padding: 0 0.4rem;
  margin: 0 0.4rem;
  background: inherit;
  color: inherit;
  border: none;
  font: inherit;
  cursor: pointer;
  outline: inherit;

  :hover:enabled {
    background-color: ",
                    "rgba(255, 255, 255, 0.25)",
                    ";
  }

  :disabled {
    color: ",
                    "#ccc",
                    ";
  }
",
                    "
  font-size: ",
                    "0.8rem",
                    ";
",
                  ],
                },
                "displayName": "Styled(styled.button)",
                "foldedComponentIds": Array [
                  "sc-jKJlTe",
                ],
                "render": [Function],
                "styledComponentId": "sc-eNQAEJ",
                "target": "button",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            onClick={[Function]}
          >
            <button
              className="bmXkwR"
              onClick={[Function]}
            >
              Hide All Hoods
            </button>
          </StyledComponent>
        </Styled(styled.button)>
      </div>
    </StyledComponent>
  </styled.div>
</Component>
`;
